src/tab.rs

```diff
@@ -1634,6 +1649,25 @@ pub struct Tab {
     search_context: Option<SearchContext>,
 }
 
+fn calculate_dir_size(path: &Path) -> u64 {
+    let mut total_size = 0;
+
+    if let Ok(entries) = fs::read_dir(path) {
+        for entry in entries.flatten() {
+            let metadata = entry.metadata().ok();
+
+            if let Some(data) = metadata {
+                if data.is_dir() {
+                    total_size += calculate_dir_size(&entry.path());
```

Comment:
Recursion is problematic as it can cause stack overflows. Consider using walkdir::WalkDir instead.